modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -1002,30 +1118,17 @@ def _union_all(
         if (
             len(other_modin_frames) == 0
             or len(self.columns) == 0
-            or all(f._has_arrow_table() for f in ([self] + other_modin_frames))
-            or any(is_string_dtype(t) for t in self._dtypes)
             or any(len(f.columns) != len(self.columns) for f in other_modin_frames)
-            or any(set(f._dtypes) != set(self._dtypes) for f in other_modin_frames)
         ):
-            return self._union_all_arrow(other_modin_frames, join, sort, ignore_index)
+            return self._union_all_arrow(frames, join, sort, ignore_index)
```

Comment:
Simply - no, because if the input frames are not materialized, the concatenation will be performed on the HDK side and will fail. The lazy concatenation will be implemented in a separate PR #5759, I'm currently working on it.